EcommerceUI/src/app/add-product/add-product.component.ts

```diff
@@ -15,6 +15,27 @@ export class AddProductComponent implements OnInit {
   product = new Product();
   widget: any;
 
+  categories : any[] = [];
```

Comment:
You should add a method to pull the categories from the DB rather than preloading them in memory like this.